src/simulation/dynamics/Thrusters/thrusterDynamicEffector/thrusterDynamicEffector.cpp

```diff
@@ -371,6 +390,39 @@ void ThrusterDynamicEffector::addThruster(THRSimConfig* newThruster, Message<SCS
 }
 
 
+/*! This method is used to update the blow down effects to the thrust and/or Isp
+* at every computeForceTorque call when the thrusters are attached to a fuel
+* tank subject to blow down effects.
+ @return void
+ */
+void ThrusterDynamicEffector::computeBlowDownDecay(THRSimConfig *currentThruster)
+{
+    THROperation *ops = &(currentThruster->ThrustOps);
+
+    if (!currentThruster->thrBlowDownCoeff.empty()) {
+        double thrustBlowDown = 0.0;
+        double thrOrder = 1.0;
+        for(auto thrCoeff = currentThruster->thrBlowDownCoeff.rbegin(); thrCoeff !=
+                                               currentThruster->thrBlowDownCoeff.rend(); thrCoeff++) {
+            thrustBlowDown += *thrCoeff * thrOrder;
+            thrOrder *= fuelMass;  // Fuel mass assigned in fuel tank's updateEffectorMassProps method
+        }
+        ops->thrustBlowDownFactor = std::clamp(thrustBlowDown / currentThruster->MaxThrust, double (0.0), double (1.0));
```

Comment:
One additional change has been made not already mentioned in these comments, based on a separate discussion between Cody and I. The thrust and Isp scaling factors for blow down have now been clamped to min = 0.0 and max = 1.0 as shown here.